packages/commerce-sdk-react/src/hooks/ShopperSearch/query.ts

```diff
@@ -19,22 +25,59 @@ the product search hit. The search result contains only products that are online
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shoppersearch.shoppersearch-1.html#productsearch} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProductSearch = (
-    arg: Argument<Client['productSearch']>
-): UseQueryResult<DataType<Client['productSearch']>, Error> => {
-    const {shopperSearch: client} = useCommerceApi()
-    return useAsync(['product-search', arg], () => client.productSearch(arg))
+function useProductSearch(
+    arg: Omit<UseProductSearchArg, 'rawResponse'> & {rawResponse?: false},
+    options?: UseQueryOptions<DataType<Client['productSearch']> | Response, Error>
+): UseQueryResult<DataType<Client['productSearch']>, Error>
+function useProductSearch(
+    arg: Omit<UseProductSearchArg, 'rawResponse'> & {rawResponse?: true},
+    options?: UseQueryOptions<DataType<Client['productSearch']> | Response, Error>
+): UseQueryResult<DataType<Client['productSearch']>, Error>
```

Comment:
```suggestion
): UseQueryResult<Response, Error>
```

For this overloaded function, `rawResponse` is true. So the query result is of type `Response` ← following what was done in ShopperProducts.

Reply:
Got it. Thanks for pointing that out. Fixed the return type in the latest commit.